From 68aa654d9f51c739cb1aac163d8aef06b3907df3 Mon Sep 17 00:00:00 2001
From: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
Date: Tue, 12 Jun 2018 02:03:16 -0500
Subject: [PATCH 3/3] acrn-bridge: improve systemd network units

To be sorted in lexical order let's add a 50- prefix to the systemd
network units files.

Now these systemd network units will be processed before the other units
prefixed by 80-. And the first (in lexical order) of the network files
that matches a given device is applied, all later files are ignored,
even if they match as well.

To be extracted all these units are inside a VM so add a Match section
to restrict them to processed inside a VM.

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
---
 tools/acrnbridge/Makefile         | 8 ++++----
 tools/acrnbridge/acrn.netdev      | 3 +++
 tools/acrnbridge/acrn.network     | 1 +
 tools/acrnbridge/acrn_tap0.netdev | 3 +++
 tools/acrnbridge/eth.network      | 1 +
 5 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/tools/acrnbridge/Makefile b/tools/acrnbridge/Makefile
index dae5c00..fe3367d 100644
--- a/tools/acrnbridge/Makefile
+++ b/tools/acrnbridge/Makefile
@@ -10,7 +10,7 @@ all:
 
 install:
 	install -d $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network
-	install -p -D -m 0644 $(OUT_DIR)/acrn.netdev $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network
-	install -p -D -m 0644 $(OUT_DIR)/acrn.network $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network
-	install -p -D -m 0644 $(OUT_DIR)/acrn_tap0.netdev $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network
-	install -p -D -m 0644 $(OUT_DIR)/eth.network $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network
+	install -p -D -m 0644 $(OUT_DIR)/acrn.netdev $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network/50-acrn.netdev
+	install -p -D -m 0644 $(OUT_DIR)/acrn.network $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network/50-acrn.network
+	install -p -D -m 0644 $(OUT_DIR)/acrn_tap0.netdev $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network/50-acrn_tap0.netdev
+	install -p -D -m 0644 $(OUT_DIR)/eth.network $(DESTDIR)/$(SYSTEMD_NETWORKDIR)/systemd/network/50-eth.network
diff --git a/tools/acrnbridge/acrn.netdev b/tools/acrnbridge/acrn.netdev
index 29283cc..67490b7 100644
--- a/tools/acrnbridge/acrn.netdev
+++ b/tools/acrnbridge/acrn.netdev
@@ -1,3 +1,6 @@
+[Match]
+Virtualization=vm
+
 [NetDev]
 Name=acrn-br0
 Kind=bridge
diff --git a/tools/acrnbridge/acrn.network b/tools/acrnbridge/acrn.network
index f06114e..620f0dd 100644
--- a/tools/acrnbridge/acrn.network
+++ b/tools/acrnbridge/acrn.network
@@ -1,5 +1,6 @@
 [Match]
 Name=e* acrn_tap*
+Virtualization=vm
 
 [Network]
 Bridge=acrn-br0
diff --git a/tools/acrnbridge/acrn_tap0.netdev b/tools/acrnbridge/acrn_tap0.netdev
index 5824cb9..fa52025 100644
--- a/tools/acrnbridge/acrn_tap0.netdev
+++ b/tools/acrnbridge/acrn_tap0.netdev
@@ -1,3 +1,6 @@
+[Match]
+Virtualization=vm
+
 [NetDev]
 Name=acrn_tap0
 Kind=tap
diff --git a/tools/acrnbridge/eth.network b/tools/acrnbridge/eth.network
index a63931b..85b46a8 100644
--- a/tools/acrnbridge/eth.network
+++ b/tools/acrnbridge/eth.network
@@ -1,5 +1,6 @@
 [Match]
 Name=acrn-br0
+Virtualization=vm
 
 [Network]
 DHCP=ipv4
-- 
2.17.1

